msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/local/TestCase1477854.java

```diff
@@ -0,0 +1,169 @@
+                        .loginHint(mLoginHint)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .speedBumpExpected(false)
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        // a) Interactive returns access token A
+        interactiveResult.assertSuccess();
+
+        // b) Fast forward device time to refresh expire interactively acquired access token but not cause access token regular expiry(in this case move by 1:35 min) - i.e. access token is now refresh expired
+        TestContext.getTestContext().getTestDevice().getSettings().forwardDeviceTime(5700);
+
+        //acquire token silently - will start a background thread to refresh access token,
+        //but in this thread the interactively acquired AT(i.e. belonging to authResult from above) will be returned.
+        final MsalAuthTestParams authTestParams2 = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(mLoginHint)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.LOGIN)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+        final MsalAuthResult firstSilentResult = msalSdk.acquireTokenSilent(authTestParams2, TokenRequestTimeout.MEDIUM);
+
+        // c) Silent returns access token A. Refresh executed In Time(eager) causes background thread to retrieve NEW access token)
+        firstSilentResult.assertSuccess();
+
+        // d) Verify interactive and silent are both = to A
+        Assert.assertTrue(interactiveResult.isAccessTokenEqual(firstSilentResult.getAccessToken()));
+
+        // e) Wait 1 minute (for background request to complete)
+        Thread.sleep(TimeUnit.MINUTES.toMillis(1));
+
+        //acquire token silently for second time, the purpose is to acquire the new AT fetched using the background thread
+        //from the above silent call
+        final MsalAuthTestParams secondSilentResult = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(mLoginHint)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.LOGIN)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+        final MsalAuthResult authResult3 = msalSdk.acquireTokenSilent(authTestParams3, TokenRequestTimeout.MEDIUM);
+        // f) Second Silent request returns access token B
+        authResult3.assertSuccess();
+
+        // g) Verify that 2nd silent request... after refresh in expired... now returns a new AT by verifying Silent 1 response not equal to silent 2 response.
+        Assert.assertFalse(firstSilentResult.isAccessTokenEqual(secondSilentResult.getAccessToken()));
+
+    }
+
+
+    @Override
```

Comment:
should these both be null? getLabUserQuery, getTempUserType
I think we throw an exception in case both of these are null.